src/TLuaInterpreter.cpp

```diff
@@ -17158,49 +17162,115 @@ void TLuaInterpreter::initIndenterGlobals()
+                                                           "%1/../3rdparty/?.lua;" // 3
+                                                           "%1/../../3rdparty/?.lua;" // 4
+                                                           "%1/../../mudlet/3rdparty/?.lua;]]" // 5
+                                                           " .. package.path")
+                      .arg(QCoreApplication::applicationDirPath()).toUtf8().constData());
+    } else {
+        luaL_dostring(pIndenterState.get(), QStringLiteral("package.path = [["
+                                                           "%1/?.lua;" // 2
+                                                           "%1/../3rdparty/?.lua;" // 3
+                                                           "%1/../../3rdparty/?.lua;" // 4
+                                                           "%1/../../mudlet/3rdparty/?.lua;]]" // 5
+                                                           " .. package.path")
+                      .arg(QCoreApplication::applicationDirPath()).toUtf8().constData());
+    }
+
+#if ! defined (Q_OS_MACOS)
+    // AppInstaller on Linux would like the search path for the binary modules
+    // to also be set to a lib sub-directory of the application directory:
+    luaL_dostring(pIndenterState.get(), QStringLiteral("package.cpath = '%1/lib/?.so;' .. package.cpath")
+                  .arg(QCoreApplication::applicationDirPath()).toUtf8().constData());
+#endif
 
-    //AppInstaller on Linux would like the search path to also be set to the current binary directory
-    luaL_dostring(pIndenterState.get(), QStringLiteral("package.cpath = package.cpath .. ';%1/lib/?.so'")
-                  .arg(QCoreApplication::applicationDirPath())
-                  .toUtf8().constData());
+// end of the #if defined(Q_OS_UNIX)
 #elif defined(Q_OS_WIN32)
-    // For Qt Creator builds, add search paths one and two levels up from here, then a 3rdparty directory:
-    luaL_dostring(pIndenterState.get(),
-                  QStringLiteral("package.path = [[%1\\?.lua;%2\\..\\3rdparty\\?.lua;%2\\..\\..\\3rdparty\\?.lua;]] .. package.path")
-                          .arg(QByteArray(LUA_DEFAULT_PATH), QDir::toNativeSeparators(QCoreApplication::applicationDirPath()))
-                          .toUtf8().constData());
+    // Insert the same function that got put into "luaGlobal.lua" in order to
+    // make the paths cleaner (and conform to what the package handler is
+    // expecting):
+    // clang-format off
+    luaL_dostring(pIndenterState.get(), R"LUA(function toNativeSeparators(rawPath)
+  if package.config:sub(1,1) == '\\' then
+    return string.gsub(rawPath, '/', '\\')
+  end
+
+  assert((package.config:sub(1,1) == '/'), "package path directory separator is neither '\\' nor '/' and cannot be handled")
+
+  return string.gsub(rawPath, '\\', '/')
+end)LUA");
+    // clang-format on
+
+    if (!QStringLiteral(LUA_DEFAULT_PATH).isEmpty()) {
```

Comment:
Isn't this exactly the same thing as above, apart from the toNativeSeparators function? Why not define the function so it handles both unix and windows variants, and then do this one time using the function? We don't need to make this super performant, it happens one time on startup. Keeping the code DRY would make it MUCH more maintainable and a lot less frustrating to read in future.

Reply:
We don't need it for non-Windows OSes because they do not - as far as I understand it - tend to use anything other than `/`as the second element of `package.config` which is the directory separator that the Lua package handler uses. This PR is trying to ease in handling alternatives for the Windows case but without changing unnecessary things for the other OS. OTOH it might be simpler to have less variations. I will take another look.

Reply:
Ah, no, it does not have the number 2 entry.

Reply:
It's not about 'does unix need it' it's about 'do we need to maintain this in multiple locations' and the answer is 'no, we do not' 

Does having case 2 in the package.path in the below case cause issues? If so, does it need to be in position 2 for everything else? If not, shove it at the end of the string you're assembling before modifying package.path. 

But I'm pretty confident the answer isn't a bunch of otherwise repeated code we have to maintain if something changes.

Reply:
I've had a go at simplifying things now - does it look any better now?

Reply:
Much better, thank you =) 
I still think the comments are perhaps a bit much, but the package.path/cpath stuff no longer sets off that twitch in my eye ;)

Reply:
They will help in the future when some poor soul is trying to work out why the files won't load on their OS - even though they think they have the right files in the right place - I had to jiggle things around to get the LCF working on FreeBSD as I recall...

... and actually the revision has made the code a bit easier to fix if something changes in the future.

As for your eye twitch - remember: "*In the country of the blind, the one-eye man will be king!*" :stuck_out_tongue_winking_eye: